pyfixest/estimation/feglm_.py

```diff
@@ -0,0 +1,520 @@
+
+        self._Y_hat_response = np.array([])
+        self.deviance = None
+        self._Xbeta = np.array([])
+
+        self._method = "feglm"
+
+    def prepare_model_matrix(self):
+        "Prepare model inputs for estimation."
+        super().prepare_model_matrix()
+
+        if self._fe is not None:
+            raise NotImplementedError("Fixed effects are not yet supported for GLMs.")
+
+        # check if Y is a weakly positive integer
+        # self._Y = _to_integer(self._Y)
+        # check for separation
+        na_separation: list[int] = []
+        if (
+            self._fe is not None
+            and self.separation_check is not None
+            and self.separation_check  # not an empty list
+        ):
+            na_separation = _check_for_separation(
+                Y=self._Y,
+                X=self._X,
+                fe=self._fe,
+                fml=self._fml,
+                data=self._data,
+                methods=self.separation_check,
+            )
+
+        if na_separation:
+            self._Y.drop(na_separation, axis=0, inplace=True)
+            self._X.drop(na_separation, axis=0, inplace=True)
+            self._fe.drop(na_separation, axis=0, inplace=True)
+            self._data.drop(na_separation, axis=0, inplace=True)
+            self._N = self._Y.shape[0]
+
+            self.na_index = np.concatenate([self.na_index, np.array(na_separation)])
+            self.n_separation_na = len(na_separation)
+
+    def to_array(self):
+        "Turn estimation DataFrames to np arrays."
+        self._Y, self._X, self._Z = (
+            self._Y.to_numpy(),
+            self._X.to_numpy(),
+            self._X.to_numpy(),
+        )
+        if self._fe is not None:
+            self._fe = self._fe.to_numpy()
+            if self._fe.ndim == 1:
+                self._fe = self._fe.reshape((self._N, 1))
+
+    def get_fit(self):
+        "Fit the GLM model via iterated weighted least squares."
+        _Y = self._Y
+        _X = self._X
+        _fe = self._fe
+        _N = self._N
+        _convergence = self.convergence  # False
+        _maxiter = self.maxiter
+        _tol = self.tol
+        _fixef_tol = self._fixef_tol
+        _solver = self._solver
+
+        # initialize
+
+        beta = np.zeros(_X.shape[1])
+        eta = np.zeros(_N)
+        mu = self._get_mu(theta=eta)
+        deviance = self._get_deviance(_Y.flatten(), mu)
+        deviance_old = deviance.copy() + 1
+
+        for r in range(_maxiter):
+            if r == 0:
+                pass
+            else:
+                converged = self._check_convergence(
+                    crit=self._get_diff(deviance=deviance, last=deviance_old),
+                    tol=_tol,
+                    r=r,
+                    maxiter=_maxiter,
+                    model=self._method,
+                )
+                if converged:
+                    break
+
+            # Step 1: _get weights w_tilde(r-1) and v(r-1) (eq. 2.5)
```

Comment:
Eq 2.5 from where? It might be a good idea to add the reference to the `get_fit` docstrings.